src/components/ArticleCard.jsx

```diff
@@ -1,18 +1,23 @@
+import { useNavigate } from "react-router-dom";
 export default function ArticleCard({ article }) {
   const date = new Date(article.created_at);
   const options = { day: "numeric", month: "long", year: "numeric" };
   const formattedDate = date.toLocaleDateString("en-GB", options);
+  const navigate = useNavigate();
 
+  function handleClick() {
+    navigate(`/articles/${article.article_id}`);
+  }
```

Comment:
Nice, good to see you using `navigate`

using a <Link> would also be fine in the actual React.